include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp

```diff
@@ -444,18 +392,225 @@ class ParticleCalorimeter : public ISimulationPlugin
+                desc,
+                masterPrefix + prefix
+            );
+            fileName.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            numBinsYaw.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            numBinsPitch.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            numBinsEnergy.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            minEnergy.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            maxEnergy.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            logScale.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            openingYaw.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            openingPitch.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            posYaw.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+            posPitch.registerHelp(
+                desc,
+                masterPrefix + prefix
+            );
+        }
+
+        void expandHelp(
+            boost::program_options::options_description & desc,
+            std::string const & masterPrefix = std::string{ }
+        )
+        {
+        }
+
+
+        void validateOptions()
+        {
+            if( notifyPeriod.size() != fileName.size() )
+                throw std::runtime_error( name + ": parameter fileName and period are not used the same number of times" );
+        }
+
+        size_t getNumPlugins() const
+        {
+            return notifyPeriod.size();
+        }
+
+        std::string getDescription() const
+        {
+            return description;
+        }
+
+        std::string getOptionPrefix() const
+        {
+            return prefix;
+        }
+
+        std::string getName() const
+        {
+            return name;
+        }
+
+        std::string const name = "ParticleCalorimeter";
+        //! short description of the plugin
+        std::string const description = "(virtually) propagates and collects particles to infinite distance";
+        //! prefix used for command line arguments
+        std::string const prefix = ParticlesType::FrameType::getName( ) + std::string( "_calorimeter" );
+    };
+
+    static std::shared_ptr< plugins::multi::IHelp > getHelp()
+    {
+        return std::shared_ptr< plugins::multi::IHelp >( new Help{ } );
+    }
+
+    ParticleCalorimeter(
+        std::shared_ptr< plugins::multi::IHelp > & help,
+        size_t const id,
+        MappingDesc* cellDescription
+    ) :
+        foldername(m_help->getOptionPrefix()),
         leftParticlesDatasetName("calorimeterLeftParticles"),
```

Comment:
missing member init fixed in:

https://github.com/ComputationalRadiationPhysics/picongpu/pull/2569/commits/896feebfff032a3f8bfad0af0fcb7cd1c7a81559